samples/event_manager/src/modules/controller.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#include "measurement_event.h"
+#include "control_event.h"
+#include "ack_event.h"
+#include "config_event.h"
+
+#define VALUE2_THRESH 15
+#define MODULE controller
+
+static bool ack_req;
+
+void send_control_event(void)
+{
+	ack_req = true;
+	struct control_event *event = new_control_event();
+
+	EVENT_SUBMIT(event);
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_measurement_event(eh)) {
+		__ASSERT_NO_MSG(ack_req != true);
+		struct measurement_event *me = cast_measurement_event(eh);
+
+		if (me->value2 >= VALUE2_THRESH
+		    || me->value2 <= -VALUE2_THRESH) {
+			send_control_event();
+		}
+
+		return false;
+	}
+
+	if (is_ack_event(eh)) {
+		__ASSERT_NO_MSG(ack_req == true);
```

Comment:
`__ASSERT_NO_MSG(ack_req);`

Reply:
also in other places where boolean is checked

btw - when checking boolean never compare to true - instead compare to false (i.e. `abc != false`). 
In C true is anything but 0 while true is define to 1, so e.g. ack_req equal to 2 will be true but it will fail the test.

Reply:
ok, fixed